### PR TITLE
feat(ingest):  add removing partition pattern in spark lineage

### PR DESCRIPTION
**Pull request by ssilb4**
in aws emr, when I use spark lineage, input table is s3 location. but It include partition location. so It doesn't match other ingestion (like glue ingestion)
Remove partition pattern. (e.g. /partition=\d+) It change database/table/partition=123 to database/table

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-integration/java/spark-lineage/README.md
Comment (jjoyce0510) [inline metadata-integration/java/spark-lineage/README.md:137]: Please add the default value! What is the default?

Comment (jjoyce0510) [reply metadata-integration/java/spark-lineage/README.md:137]: Also what are sample values?

File: metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/HdfsPathDataset.java
Comment (jjoyce0510) [inline metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/HdfsPathDataset.java:17]: If we returned null in previous version we can check whether its not equal to null instead of empty string.

**Review by jjoyce0510 — CHANGES_REQUESTED** [review]
Once changes are requested, we will re-review

**Review by jjoyce0510 — APPROVED** [review]
LGTM. Waiting on final review from @treff7es

File: metadata-integration/java/spark-lineage/README.md
Comment (ssilb4) [reply metadata-integration/java/spark-lineage/README.md:137]: I don't think it need to have default value. If it is null, It don't change any value.
I thought description is sample value, should I do more? (value: /partition=\d+)

File: metadata-integration/java/spark-lineage/src/main/java/datahub/spark/DatasetExtractor.java
Comment (ssilb4) [reply metadata-integration/java/spark-lineage/src/main/java/datahub/spark/DatasetExtractor.java]: I changed

File: metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/HdfsPathDataset.java
Comment (ssilb4) [reply metadata-integration/java/spark-lineage/src/main/java/datahub/spark/model/dataset/HdfsPathDataset.java]: I changed

**Comment by jjoyce0510** [comment]
Looks like build is failing. Going to update to the latest..

**Comment by ssilb4** [comment]
hmm.. should I fix more? I don't understand why smoke test failed.

**Review by treff7es — APPROVED** [review]
lgtm

**Pull request merged**
